File: libraries/go/storage1024.go
package storage1024

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"mime/multipart"
	"net/http"
	"os"
)

type Storage1024 struct {
	APIBase string
	Token   string
	UserID  string
}

func New() *Storage1024 {
	return &Storage1024{
		APIBase: "https://storage1024.onrender.com/api",
	}
}

func (s *Storage1024) SetToken(token string) {
	s.Token = token
}

func (s *Storage1024) SetUserID(userID string) {
	s.UserID = userID
}

func (s *Storage1024) GetGV(name string) (string, error) {
	url := fmt.Sprintf("%s/projects/%s/gv/%s", s.APIBase, s.UserID, name)
	req, _ := http.NewRequest("GET", url, nil)
	req.Header.Set("Authorization", "Bearer "+s.Token)

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		return "", err
	}
	defer resp.Body.Close()

	var result struct {
		Value string `json:"value"`
	}
	json.NewDecoder(resp.Body).Decode(&result)
	return result.Value, nil
}

func (s *Storage1024) AddGV(name, value string) error {
	url := fmt.Sprintf("%s/projects/%s/gv", s.APIBase, s.UserID)
	data := map[string]string{"alias": name, "value": value}
	jsonData, _ := json.Marshal(data)

	req, _ := http.NewRequest("POST", url, bytes.NewBuffer(jsonData))
	req.Header.Set("Authorization", "Bearer "+s.Token)
	req.Header.Set("Content-Type", "application/json")

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		return err
	}
	defer resp.Body.Close()
	return nil
}

func (s *Storage1024) UploadFile(alias, filePath string) error {
	url := fmt.Sprintf("%s/projects/%s/upload", s.APIBase, s.UserID)
	file, err := os.Open(filePath)
	if err != nil {
		return err
	}
	defer file.Close()

	body := &bytes.Buffer{}
	writer := multipart.NewWriter(body)
	part, _ := writer.CreateFormFile("file", filePath)
	io.Copy(part, file)
	writer.WriteField("alias", alias)
	writer.Close()

	req, _ := http.NewRequest("POST", url, body)
	req.Header.Set("Authorization", "Bearer "+s.Token)
	req.Header.Set("Content-Type", writer.FormDataHeader())

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		return err
	}
	defer resp.Body.Close()
	return nil
}

File: libraries/php/Storage1024.php
<?php

namespace Storage1024;

class Storage1024 {
    private string $apiBase = "https://storage1024.onrender.com/api";
    private string $token = "";
    private string $userID = "";

    public function setToken(string $token): void {
        $this->token = $token;
    }

    public function setUserID(string $userID): void {
        $this->userID = $userID;
    }

    public function getGV(string $name): string {
        $url = "{$this->apiBase}/projects/{$this->userID}/gv/{$name}";
        $options = [
            'http' => [
                'header' => "Authorization: Bearer {$this->token}\r\n"
            ]
        ];
        $context = stream_context_create($options);
        $response = file_get_contents($url, false, $context);
        $data = json_decode($response, true);
        return $data['value'] ?? "";
    }

    public function addGV(string $name, string $value): bool {
        $url = "{$this->apiBase}/projects/{$this->userID}/gv";
        $data = json_encode(['alias' => $name, 'value' => $value]);
        $options = [
            'http' => [
                'header' => "Authorization: Bearer {$this->token}\r\n" .
                            "Content-Type: application/json\r\n",
                'method' => 'POST',
                'content' => $data
            ]
        ];
        $context = stream_context_create($options);
        $response = file_get_contents($url, false, $context);
        return $response !== false;
    }

    public function uploadFile(string $alias, string $filePath): bool {
        if (!file_exists($filePath)) return false;

        $url = "{$this->apiBase}/projects/{$this->userID}/upload";
        $curl = curl_init();
        
        $cFile = new \CURLFile($filePath);
        $postData = [
            'file' => $cFile,
            'alias' => $alias
        ];

        curl_setopt($curl, CURLOPT_URL, $url);
        curl_setopt($curl, CURLOPT_POST, true);
        curl_setopt($curl, CURLOPT_POSTFIELDS, $postData);
        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($curl, CURLOPT_HTTPHEADER, [
            "Authorization: Bearer {$this->token}"
        ]);

        $response = curl_exec($curl);
        curl_close($curl);
        
        return $response !== false;
    }
}
